File: workspace_js/services/js-executor/src/modules/scripts/interfaces/http/routes/internal_routes.ts
import type { ScriptServicePort } from '@modules/scripts/application/ports';

import { Router } from '@mapexos/microservices';
import {
	getTemplateScriptTestInternal,
	getSamplePayloadInternal,
} from '@modules/scripts/interfaces/http/handlers/internal_handler';

/**
 * Registers internal API routes for service-to-service communication.
 *
 * These routes are protected by API Key authentication and are used internally
 *
 * Routes:
 * - GET /:orgId/:templateId/script_test - Fetch raw scriptTest from template
 *   - orgId: "mapexos_public" for system templates, or org's ID for private templates
 *   - templateId: The MongoDB ObjectID of the template
 *   - Returns: { scriptTest: object } - The raw scriptTest payload
 *
 * - GET /:orgId/:templateId/sample_payload - Generate processed sample payload
 *   - orgId: "mapexos_public" for system templates, or org's ID for private templates
 *   - templateId: The MongoDB ObjectID of the template
 *   - Returns: { samplePayload: object } - The processed/standardized payload
 *   - Used by Rule Test Runner UI to get an editable sample payload
 *
 * @param service - The script service port interface
 * @returns {Router} A Router instance containing the registered internal routes.
 */
export function registerInternalRoutes(service: ScriptServicePort): Router {
	const group = Router();

	/** Get raw scriptTest from template using orgId + templateId (TieredCache key format) */
	group.get('/:orgId/:templateId/script_test', getTemplateScriptTestInternal(service));

	/** Get processed sample payload by executing template scripts against scriptTest */
	group.get('/:orgId/:templateId/sample_payload', getSamplePayloadInternal(service));

	return group;
}

File: workspace_go/services/assets/src/modules/assettemplates/interfaces/http/routes/internal_routes.go
package routes

import (
	web "github.com/Mapex-Solutions/mapexGoKit/microservices/http/web"

	"assets/src/modules/assettemplates/application/ports"
	"assets/src/modules/assettemplates/interfaces/http/handlers"
)

// RegisterInternalRoutes registers internal API routes for TieredCache fallback.
//
// These routes are protected by API Key authentication and are used internally
// by other services (JS-Executor, Events) when L2 cache miss occurs.
//
// Routes:
//   - GET /:templateId - Fetch template and repopulate L2 cache
//
// Parameters:
//   - group: The Fiber router group for internal routes (e.g., /internal/templates)
//   - service: The AssetTemplateServicePort interface for business operations
func RegisterInternalRoutes(group web.Router, service ports.AssetTemplateServicePort) {
	group.Get("/:templateId", handlers.GetTemplateForCacheFallback(service))
}
